src/js/services/conversation-service.js

```diff
@@ -0,0 +1,76 @@
+import { store } from 'stores/app-store';
+import { messageAdded, setConversation } from 'actions/conversation-actions';
+import { setFayeSubscription, unsetFayeSubscription } from 'actions/faye-actions';
+import { core } from 'services/core';
+import { immediateUpdate } from 'services/user-service';
+import { initFaye } from 'utils/faye';
+
+export function sendMessage(text) {
+    var sendFn = () => {
+        const message = {
+            text: text,
+            role: 'appUser'
+        };
+
+        store.dispatch(messageAdded(message));
```

Comment:
So we would optimistically create the message and then if it's rejected by the server, we'd show an error message?